scripts/irods/test/test_dynamic_peps.py

```diff
@@ -322,3 +323,116 @@ def test_pep_database_data_object_finalize__issue_6385(self):
+
+                # Add a rule to core.re which when triggered will add the vector size and elements
+                # of "*values" to the session collection as AVUs.
+                with open(core_re_path, 'a') as core_re:
+                    core_re.write(dedent('''
+                    pep_database_execute_genquery2_sql_pre(*inst, *ctx, *out, *sql, *values, *output) {{
+                        msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{attr_name_1}', *values."0", '');
+                        msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{attr_name_2}', *values."size", '');
+                    }}
+                    '''.format(**locals())))
+
+                try:
+                    # Trigger PEP, ignore output.
+                    self.user.assert_icommand(['iquery', f"select COLL_NAME where COLL_NAME = '{attr_value_1}'"], 'STDOUT')
+
+                    # Show the session collection has an AVU attached to it.
+                    # This is only possible due to vector of strings being serializable.
+                    out, _, _ = self.user.run_icommand(['imeta', 'ls', '-C', self.user.session_collection])
+                    self.user.assert_icommand(['imeta', 'ls', '-C', self.user.session_collection], 'STDOUT', [
+                        f'attribute: {attr_name_1}\nvalue: {attr_value_1}\n',
+                        # The value is 2 due to the parser adding additional bind variables.
+                        # For this specific case, a second bind variable for the username is included.
+                        f'attribute: {attr_name_2}\nvalue: 2\n'
+                    ])
+
+                finally:
+                    # This isn't required, but we do it here to keep the database metadata
+                    # tables small, for performance reasons.
+                    self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, attr_name_1, attr_value_1])
+                    self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, attr_name_2, '2'])
+
+    @unittest.skipIf(plugin_name != 'irods_rule_engine_plugin-irods_rule_language' or test.settings.RUN_IN_TOPOLOGY, 'Requires NREP in single node zone.')
+    def test_if_vector_of_strings_is_exposed__issue_7570(self):
+        config = IrodsConfig()
+
+        with lib.file_backed_up(config.server_config_path):
+            try:
+                # Lower the delay server's sleep time so that rule are executed quicker.
+                config.server_config['advanced_settings']['delay_server_sleep_time_in_seconds'] = 1
+                lib.update_json_file_from_dict(config.server_config_path, config.server_config)
+
+                # Restart so the new server configuration takes effect.
+                IrodsController().restart(test_mode=True)
+
+                core_re_path = os.path.join(config.core_re_directory, 'core.re')
+
+                with lib.file_backed_up(core_re_path):
```

Comment:
This is fine as is, but there is something called `core_file.temporary_core_file` that kind of does what you're doing here. Don't have to use it, but figured I'd throw it out there.

Reply:
Oh right. I've used that a few times before. I'll update this to use the newer style since it improves readability.

Reply:
Done.